index.js

```diff
@@ -10,8 +10,42 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+const fetchColors = async ({ name, hex, compName, compHex }) => {
+  try {
+    const response = await fetch(COLORS);
+    if (!response.ok) {
+      throw new Error(`Response status: ${response.status}`);
+    }
+
+    const json = await response.json();
+    if (name) {
+      const lowerCaseName = name.toLowerCase();
+      const result = json.filter(
+        (color) => color.name.toLowerCase().includes(lowerCaseName)
+      );
+      return result;
+    } else if (hex) {
```

Comment:
Remember to DRY your code. The string comparison could be moved to its own function so if the requirement changed to be case sensitive you only need to do it in one place.